specs/eventing/control-plane.md

```diff
@@ -0,0 +1,985 @@
+[event delivery as described below](#event-delivery). The Channel MAY also
+forward events to associated Subscriptions for with the `Ready` condition is not
+currently `true`. (One example: a Subscription which is in the process of being
+programmed in the Channel data plane might receive _some_ events before the data
+plane programming was complete and the Subscription was updated to set the
+`Ready` condition to `true`.)
+
+If multiple Subscriptions with the same destination are associated with the same
+Channel, each Subscription MUST generate one delivery attempt per Subscription.
+The implementation MAY attach additional event attributes or other metadata
+distinguishing between these deliveries. The implementation MUST NOT modify the
+event payload in this process.
+
+### Event Delivery
+
+Once a Trigger or Subscription has decided to deliver an event, it MUST do the
+following:
+
+1. Resolve all URLs and delivery options, using the values in `status` for URL
+   resolution.
+
+1. Attempt delivery to the `status.subscriberUri` URL following the
+   [data plane contract](./data-plane.md).
+
+   1. If the event delivery fails with a retryable error, it SHOULD be retried
+      up to `retry` times, following the `backoffPolicy` and `backoffDelay`
+      parameters if specified.
+
+1. If the delivery attempt is successful (either the original request or a
+   retry) and no event is returned, the event delivery is complete.
+
+1. If the delivery attempt is successful (either the original request or a
+   retry) and an event is returned in the reply, the reply event will be
+   delivered to the `status.replyUri` destination (for Subscriptions) or added
+   to the Broker for processing (for Triggers). If `status.replyUri` is not
+   present in the Subscription, the reply event is be dropped.
+
+   1. If delivery of the reply event fails with a retryable error, the delivery
+      of the reply event to the reply destination SHOULD be retried up to
+      `retry` times, following the `backoffPolicy` and `backoffDelay` parameters
+      if specified.
+
+1. If all retries are exhausted for either the original delivery or the retry,
+   or if a non-retryable error is received, the event should be delivered to the
+   `deadLetterSink` in the delivery options. If no `deadLetterSink` is
+   specified, the event is dropped.
+
+   The implementation MAY set additional attributes on the event or wrap the
+   failed event in a "failed delivery" event; this behavior is not (currently)
+   standardized.
+
+   If delivery of the reply event fails with a retryable error, the delivery to
+   the `deadLetterSink` SHOULD be retried up to `retry` times, following the
+   `backoffPolicy` and `backoffDelay` parameters if specified. Alternatively,
+   implementations MAY use an equivalent internal mechanism for delivery (for
+   example, if the `ref` form of `deadLetterSink` points to a compatible
+   implementation).
+
+## Detailed Resources
```

Comment:
Can we get some text to make it clear that the RFC2119 keywords in the last column indicate whether an impl MUST support it or not. While the keyword in the 2nd column is an indication of whether it is REQUIRED for the user to specify it or not. And what happens if the impl doesn't have to support it but the user includes it? Error or silently ignore?

Reply:
Regardless of the joker who wrote it :-)  I don't think it's clear that "delivery" is optional for a client but required for a server for a server to support it.

Reply:
It seems like it would easier to just mandate that all servers must support these fields. It'll help interop.